generator/mavgen.py

```diff
@@ -101,31 +142,12 @@ def mavgen_validate(xmlfile):
         xml.append(mavparse.MAVXML(fname, opts.wire_protocol))
 
     # expand includes
-    for x in xml[:]:
-        for i in x.include:
-            fname = os.path.join(os.path.dirname(x.filename), i)
-
-            # Validate XML file with XSD file if possible.
-            if opts.validate:
-                print("Validating %s" % fname)
-                if not mavgen_validate(fname):
-                    return False
-            else:
-                print("Validation skipped for %s." % fname)
-
-            # Parsing
-            print("Parsing %s" % fname)
-            xml.append(mavparse.MAVXML(fname, opts.wire_protocol))
-
-            # include message lengths and CRCs too
-            x.message_crcs.update(xml[-1].message_crcs)
-            x.message_lengths.update(xml[-1].message_lengths)
-            x.message_min_lengths.update(xml[-1].message_min_lengths)
-            x.message_flags.update(xml[-1].message_flags)
-            x.message_target_system_ofs.update(xml[-1].message_target_system_ofs)
-            x.message_target_component_ofs.update(xml[-1].message_target_component_ofs)
-            x.message_names.update(xml[-1].message_names)
-            x.largest_payload = max(x.largest_payload, xml[-1].largest_payload)
+    for i in range(5):
```

Comment:
FYI Max 5 levels of nesting hard coded. We could put as a value up top of file, but I figure this is most we'll need for foreseeable future.